ament_tools/build_types/cmake.py

```diff
@@ -188,13 +216,62 @@ def _common_cmake_on_build(self, should_run_configure, context, prefix, extra_cm
                         env['CL'] = '/MP'
             cmd += [
                 '/p:Configuration=%s' %
-                self._get_visual_studio_configuration(context), solution_file]
+                self._get_configuration_from_cmake(context), solution_file]
             yield BuildAction(cmd, env=env)
+        elif IS_MACOSX:
+            if self._using_xcode_generator(context):
+                if XCODEBUILD_EXECUTABLE is None:
+                    raise VerbExecutionError("Could not find 'xcodebuild' executable")
```

Comment:
Just based on the code, I think this only happens when you've specified that you want to use the xcode generator.